syft/frameworks/torch/tensors/interpreters/additive_shared.py

```diff
@@ -116,6 +117,11 @@ def __init__(
         # max value for shares in field
         self._max_value = None
 
+        if crypto_provider is None:
+            warnings.warn(
+                "You haven't provided crypto_provider. You won't be able to perform certain operations like multiplication"
+            )
+
         self.crypto_provider = (
             crypto_provider if crypto_provider is not None else sy.hook.local_worker
```

Comment:
You can remove this:
```
self.crypto_provider = (
            crypto_provider if crypto_provider is not None else sy.hook.local_worker	            crypto_provider if crypto_provider is not None else sy.hook.local_worker
```
to keep crypto_provider  = None otherwise your next check won't work